### PR TITLE
MWPW-128373 - Adding a confirmation modal(pop-up) for fg promote step.

### PR DESCRIPTION
**Pull request by sirivuri**
- Adds a pop-up for users to confirm before promoting the content.

Resolves: [MWPW-128373](https://jira.corp.adobe.com/browse/MWPW-128373)

Screenshot:
Screen Shot 2023-04-04 at 10 45 16 AM

**Test URLs:**

- Before: https://main--milo--adobecom.hlx.page/tools/floodgate/index.html?project=milo--adobecom&referrer=https%3A%2F%2Fadobe.sharepoint.com%2F%3Ax%3A%2Fr%2Fsites%2Fadobecom%2F_layouts%2F15%2FDoc.aspx%3Fsourcedoc%3D%257BB3FB4BFF-8B65-4D13-B4B1-C5AF42AAB315%257D%26file%3DBook.xlsx%26action%3Ddefault%26mobileredirect%3Dtrue

- After: https://main--milo--sirivuri.hlx.page/tools/floodgate/index.html?project=milo--adobecom&referrer=https%3A%2F%2Fadobe.sharepoint.com%2F%3Ax%3A%2Fr%2Fsites%2Fadobecom%2F_layouts%2F15%2FDoc.aspx%3Fsourcedoc%3D%257BB3FB4BFF-8B65-4D13-B4B1-C5AF42AAB315%257D%26file%3DBook.xlsx%26action%3Ddefault%26mobileredirect%3Dtrue

### CODE REVIEW

**Review by sukamat — COMMENTED** [review]
@sirivuri  Looks good. Few things to update...
* The confirmation dialog is required for both copy and promote.
* It will be nice to have a more relevant messaging for the author per action as to what it will do and if they want to continue with it.
* Adjust the size of the dialog to be smaller or better yet fit the width of the content in the dialog with some padding.

**Comment by honstar** [comment]
I agree with @sukamat : the dialog message should indicate the FG operation that is to be executed.

**Comment by meganthecoder** [comment]
Do you have the eslint and stylint extensions set up for vscode? I'm seeing a couple things that should've gotten flagged by the linters

**Comment by sirivuri** [comment]
@meganthecoder  pushed a few more changes with eslint recommendations.   @sukamat

File: tools/floodgate/css/floodgate.css [hunk 1]
```diff
@@ -12,3 +12,28 @@
...
+.fg-modal {
+  display: none;
+  position: fixed;
+  z-index: 1;
+  left: 0;
+  top: 0;
+  width: 100%;
+  height: 100%;
+  overflow: auto;
+  background-color: rgba(0, 0, 0, 0.5);
```
Comment (meganthecoder) [inline tools/floodgate/css/floodgate.css:25]: Did you add stylelint as well? This should be:
```suggestion
  background-color: rgb(0 0 0 / 50%);
```

File: tools/floodgate/js/floodgate.js [hunk 2]
```diff
@@ -15,6 +15,8 @@ import {
 import promoteFloodgatedFiles from './promote.js';
 import floodgateContent from './copy.js';
 
+const modal = document.getElementById('fg-modal');
```
Comment (meganthecoder) [inline tools/floodgate/js/floodgate.js:18]: You should avoid putting variables in the global scope. If `modal` is only needed in `setListeners`, then it should live there

File: tools/floodgate/js/floodgate.js [hunk 3]
```diff
@@ -40,8 +42,17 @@ async function refreshPage(config, projectDetail, project) {
 
 function setListeners(project, projectDetail) {
   document.querySelector('#reloadProject button').addEventListener('click', reloadProject);
-  document.querySelector('#copyFiles button').addEventListener('click', () => floodgateContent(project, projectDetail));
-  document.querySelector('#promoteFiles button').addEventListener('click', () => promoteFloodgatedFiles(project));
+  document.querySelector('#copyFiles button').addEventListener('click', (e) => {
+    modal.getElementsByTagName('p')[0].innerText = `Confirm to ${e.target.textContent}`;
+    document.querySelector('#fg-modal #yes-btn').addEventListener('click', () => { modal.style.display = 'none'; floodgateContent(project, projectDetail); });
+    modal.style.display = 'block';
+  });
```
Comment (meganthecoder) [inline tools/floodgate/js/floodgate.js:49]: I don't love that you're setting an event listener within another listener. Is there a reason why you can't set it outside?

File: tools/floodgate/js/floodgate.js [hunk 4]
```diff
@@ -40,8 +42,17 @@ async function refreshPage(config, projectDetail, project) {
 
 function setListeners(project, projectDetail) {
   document.querySelector('#reloadProject button').addEventListener('click', reloadProject);
-  document.querySelector('#copyFiles button').addEventListener('click', () => floodgateContent(project, projectDetail));
-  document.querySelector('#promoteFiles button').addEventListener('click', () => promoteFloodgatedFiles(project));
+  document.querySelector('#copyFiles button').addEventListener('click', (e) => {
+    modal.getElementsByTagName('p')[0].innerText = `Confirm to ${e.target.textContent}`;
+    document.querySelector('#fg-modal #yes-btn').addEventListener('click', () => { modal.style.display = 'none'; floodgateContent(project, projectDetail); });
```
Comment (meganthecoder) [inline tools/floodgate/js/floodgate.js:47]: These should go on separate lines
```suggestion
    document.querySelector('#fg-modal #yes-btn').addEventListener('click', () => {
      modal.style.display = 'none';
      floodgateContent(project, projectDetail);
    });
```

**Comment by sirivuri** [comment]
@meganthecoder  Using the same modal pop-up for both copy and promote actions. Thought the easier way to pass these actions on the confirmation is to loop in these event handlers.

File: tools/floodgate/js/floodgate.js [hunk 5]
```diff
@@ -39,9 +39,23 @@ async function refreshPage(config, projectDetail, project) {
...
+    });
+    modal.style.display = 'block';
+  });
+  document.querySelector('#promoteFiles button').addEventListener('click', (e) => {
+    modal.getElementsByTagName('p')[0].innerText = `Confirm to ${e.target.textContent}`;
+    document.querySelector('#fg-modal #yes-btn').addEventListener('click', () => {
+      modal.style.display = 'none'; promoteFloodgatedFiles(project);
+    });
+    modal.style.display = 'block';
+  });
```
Comment (meganthecoder) [inline tools/floodgate/js/floodgate.js:57]: I see what you're saying. I hadn't noticed that it was the same modal button. In that case, I guess it's fine to add the listeners within the other like you're doing, but you really should be removing them. It's best practice to remove listeners when you don't need them anyway. The code below is what I'd do.

Also, you'll notice that the variables should go at the top of the block, the event listeners should go at the bottom of their blocks, and anything with a semicolon should be on it's own line

```
  const handleFloodgateConfirm = ({ target }) => {
    modal.style.display = 'none';
    floodgateContent(project, projectDetail);
    target.removeEventListener('click', handleFloodgateConfirm);
  };
  const handlePromoteConfirm = ({ target }) => {
    modal.style.display = 'none';
    promoteFloodgatedFiles(project);
    target.removeEventListener('click', handlePromoteConfirm);
  };
  ...
  document.querySelector('#copyFiles button').addEventListener('click', (e) => {
    modal.getElementsByTagName('p')[0].innerText = `Confirm to ${e.target.textContent}`;
    modal.style.display = 'block';
    document.querySelector('#fg-modal #yes-btn').addEventListener('click', handleFloodgateConfirm);
  });
  document.querySelector('#promoteFiles button').addEventListener('click', (e) => {
    modal.getElementsByTagName('p')[0].innerText = `Confirm to ${e.target.textContent}`;
    modal.style.display = 'block';
    document.querySelector('#fg-modal #yes-btn').addEventListener('click', handlePromoteConfirm);
  });
```

Comment (meganthecoder) [reply tools/floodgate/js/floodgate.js:57]: If you're curious, a lot of these points come from the airbnb style guide, which our linter rules are based on https://github.com/airbnb/javascript

Comment (sirivuri) [reply tools/floodgate/js/floodgate.js]: Thanks @meganthecoder. Added these changes now.

**Review by meganthecoder — APPROVED** [review]

**Review by honstar — APPROVED** [review]

**Pull request merged**